src/dataspec/base.py

```diff
@@ -150,77 +150,144 @@ def tag(self) -> Tag:  # pragma: no cover
+        Validate the value ``v`` against the Spec, throwing a
+        :py:class:`dataspec.ValidationError` containing a list of all of the Spec
+        failures for ``v`` , if any. Returns :py:obj:`None` otherwise.
 
-        Throws a ``ValidationError`` with a list of Spec failures, if any.
+        :param v: a value to validate
+        :return: :py:obj:`None`
         """
         errors = list(self.validate(v))
         if errors:
             raise ValidationError(errors)
 
     def is_valid(self, v: Any) -> bool:
-        """Returns True if ``v`` is valid according to the Spec, otherwise returns
-        False."""
+        """
+        Returns :py:obj:`True` if ``v`` is valid according to the Spec, otherwise
+        returns :py:obj:`False` .
+
+        :param v: a value to validate
+        :return: :py:obj:`True` if the value is valid according to the Spec, otherwise
+            :py:obj:`False`
+        """
         try:
             next(self.validate(v))
         except StopIteration:
             return True
         else:
             return False
 
-    def _default_conform(self, v):
+    @property
+    def conformer(self) -> Optional[Conformer]:
+        """Return the custom conformer attached to this Spec, if one is defined."""
+        return None
+
+    def conform(self, v: Any):
         """
-        Default conformer for the Spec.
+        Conform ``v`` to the Spec, returning the possibly conformed value or an
+        instance of :py:class:`dataspec.Invalid` if the value cannot be conformed.
 
-        If no custom conformer is specified, this conformer will be invoked.
+        :param v: a value to conform
+        :return: a conformed value or a :py:class:`dataspec.Invalid` instance if the
+            input value could not be conformed
         """
-        return v
+        if self.is_valid(v):
+            return self.conform_valid(v)
+        else:
+            return INVALID
 
-    def conform_valid(self, v):
+    def conform_valid(self, v: Any):
         """
         Conform ``v`` to the Spec without checking if v is valid first and return the
         possibly conformed value or ``INVALID`` if the value cannot be conformed.
 
         This function should be used only if ``v`` has already been check for validity.
-        """
-        return self.conformer(v)
 
-    def conform(self, v: Any):
-        """Conform ``v`` to the Spec, first checking if v is valid, and return the
-        possibly conformed value or ``INVALID`` if the value cannot be conformed."""
-        if self.is_valid(v):
-            return self.conform_valid(v)
-        else:
+        :param v: a *validated* value to conform
+        :return: a conformed value or a :py:class:`dataspec.Invalid` instance if the
+            input value could not be conformed
+        """
+        if self.conformer is None:
+            return v
+        try:
+            return self.conformer(v)  # pylint: disable=not-callable
+        except Exception:
```

Comment:
I think you could end up in a situation where you use conform valid and you think your input failed validation but actually your conformer was erroring out

Reply:
That's a fair point. I think it was probably me trying to address two different and inter-related issues at the same time and getting mixed up. To be fair, you could call `Spec.conformer(value)` yourself to get your exception, but that's not super clear. 

I'm thinking we should probably move the `try` block into `Spec.conform` and leave `Spec.conform_valid` as it was.

Reply:
I removed the entire `try` block, because the more I thought about it, the more I agreed that it doesn't ever make sense to swallow exceptions. Thanks!